packages/core-data/src/entity-provider.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { defaultEntities, kinds } from './entities';
+
+const _entities = {
+	...defaultEntities.reduce( ( acc, entity ) => {
+		if ( ! acc[ entity.kind ] ) {
+			acc[ entity.kind ] = {};
+		}
+		acc[ entity.kind ][ entity.name ] = { context: createContext() };
+		return acc;
+	}, {} ),
+	...kinds.reduce( ( acc, kind ) => {
+		acc[ kind.name ] = {};
+		return acc;
+	}, {} ),
+};
+const getEntity = ( kind, type ) => {
+	if ( ! _entities[ kind ] ) {
+		throw new Error( `Missing entity config for kind: ${ kind }.` );
+	}
```

Comment:
If we remove this check, and just create an empty object, the first time a "kind" is requested, we can simplify furthere (remove the `_entities` initialization entirely. I'm not certain it  adds much value  to  try to validate that it's an existing kind because we can also just pass a random entity "type" and it will pass the test (we don't validate entity names).

Reply:
Because `type` can be dynamic. `kind` cannot, you need an entity config for any of the API to work.

Reply:
I don't see why "kind" can't be dynamic in the future as well tbh. I still think it's . just useless code but I'm fine shipping that way.